### PR TITLE
Update to ucfg v0.1.1

### PR DESCRIPTION
**Pull request by urso**
Upgrading ucfg from v0.0.0 to v0.1.1, see [changelog](https://github.com/urso/ucfg/blob/master/CHANGELOG.md)

Fix vendoring problem by shadowing *ucfg.Config via *common.Config.

Use time.Duration parsing support in ucfg for some duration types in libbeat and packetbeat. time.Duration parsing will treat raw number (int, float) as seconds and string will be parsed by time.ParseDuration -> change if backwards compatible to code places using 'int' for durations in seconds.

Note: Can not make use of regexp parsing support in ucfg, due to ucfg using regexp.Compile but beats using regexp.CompilePOSIX (which unfortunately works a little differently, potentially breaking backwards compatibility, see: http://play.golang.org/p/soMR8CXcwy)

Note: In general with ucfg when unpacking, the object to unpack into should have default values already be initialized instead of using pointers and do post-unpacking checks.

prefer:

```
    config := struct { number int }{ number: 5}
    err := cfg.Unpack(&config)
```

over 

```
    config := struct { number *int }
    err := cfg.Unpack(&config)
    ...
    var number int
    if config.number != nil {
        number = *config.number
    } else {
        number = 5
    }
```

### CODE REVIEW

File: libbeat/beat/beat.go [hunk 1]
```diff
@@ -28,14 +28,13 @@ import (
...
-	"github.com/satori/go.uuid"
-	"github.com/urso/ucfg"
-
 	"github.com/elastic/beats/libbeat/cfgfile"
+	"github.com/elastic/beats/libbeat/common"
 	"github.com/elastic/beats/libbeat/filter"
 	"github.com/elastic/beats/libbeat/logp"
 	"github.com/elastic/beats/libbeat/publisher"
 	"github.com/elastic/beats/libbeat/service"
+	"github.com/satori/go.uuid"
```
Comment (ruflin) [inline libbeat/beat/beat.go]: We should probably write down what rule we are following here. So far I tried to do:

* go shipped packages
* external packages from any source
* internal packages

WDYT?

File: libbeat/publisher/publish.go [hunk 2]
```diff
@@ -78,12 +76,12 @@ type PublisherType struct {
...
 type ShipperConfig struct {
-	common.EventMetadata  `config:",inline"` // Fields and tags to add to each event.
-	Name                  string
-	Refresh_topology_freq int
-	Ignore_outgoing       bool
-	Topology_expire       int
-	Geoip                 common.Geoip
+	common.EventMetadata `config:",inline"` // Fields and tags to add to each event.
+	Name                 string
+	RefreshTopologyFreq  time.Duration `config:"regresh_topology_freq"`
```
Comment (ruflin) [inline libbeat/publisher/publish.go]: s/regresh/refresh/

File: vendor/github.com/urso/ucfg/error.go [hunk 3]
```diff
@@ -0,0 +1,261 @@
...
+
+	ErrArraySizeMistach = errors.New("Array size mismatch")
+
+	ErrExpectedObject = errors.New("expected object")
+
+	ErrNilConfig = errors.New("config is nil")
+
+	ErrNilValue = errors.New("nil value is invalid")
+
+	ErrTODO = errors.New("TODO - implement me")
```
Comment (andrewkroh) [inline vendor/github.com/urso/ucfg/error.go]: Looks like this is no longer used and can be removed in your next batch of ucfg changes.

File: packetbeat/protos/amqp/config.go [hunk 4]
```diff
@@ -16,7 +16,7 @@ type amqpConfig struct {
 var (
 	defaultConfig = amqpConfig{
 		ProtocolCommon: config.ProtocolCommon{
-			TransactionTimeout: protos.DefaultTransactionTimeout,
+			TransactionTimeout: protos.DefaultTransactionExpiration,
```
Comment (ruflin) [inline packetbeat/protos/amqp/config.go]: Was that renamed?

**Comment by ruflin** [comment]
LGTM. See comments above. Didn't review the changes in ucfg. @urso we should also move the PR's in ucfg.

File: libbeat/beat/beat.go (hunk 1 above)
Comment (urso) [inline libbeat/beat/beat.go]: me too, but goimports did always change it back to this, I guess it sorts alphabetically?

File: libbeat/publisher/publish.go (hunk 2 above)
Comment (urso) [inline libbeat/publisher/publish.go]: ouch

File: packetbeat/protos/amqp/config.go (hunk 4 above)
Comment (urso) [inline packetbeat/protos/amqp/config.go]: no, there are 2 variables in protos.

File: libbeat/beat/beat.go (hunk 1 above)
Comment (ruflin) [inline libbeat/beat/beat.go]: So far I assumed, if there is a new line inbetween, it should not get messed up?

File: packetbeat/protos/amqp/config.go (hunk 4 above)
Comment (ruflin) [inline packetbeat/protos/amqp/config.go]: Is there a reason you switched? Do they have the same content?

Comment (urso) [inline packetbeat/protos/amqp/config.go]: different type, used to be int but is time.Duration now.

File: libbeat/beat/beat.go (hunk 1 above)
Comment (urso) [inline libbeat/beat/beat.go]: me too, but for some reason there is a problem with github.com/satori/go.uuid . I'm happy it's not automatically removed anymore (we have had PR's removing import of this lib in past due to goimports)

Comment (ruflin) [inline libbeat/beat/beat.go]: perhaps because of the `go.` part?

**Comment by ruflin** [comment]
@urso you missed a change in the apache module of metricbeat. I assume you rebased and this change sneaked in yesterday evening.

File: vendor/github.com/urso/ucfg/error.go (hunk 3 above)
Comment (urso) [inline vendor/github.com/urso/ucfg/error.go]: true, feel free to create PR for ucfg ;)

File: packetbeat/protos/amqp/config.go (hunk 4 above)
Comment (urso) [inline packetbeat/protos/amqp/config.go]: removed DefaultTransactionTimeout variable + found TransactionTimeout being used kinda inconsistently by default (some modules treat as seconds, others as milliseconds in config file)

**Comment by urso** [comment]
yup, found fixed the metricsbeat one

**Comment by ruflin** [comment]
@urso LGTM. You want to squash?

**Comment by urso** [comment]
@ruflin done

**Pull request merged**